debian/ubuntu-advantage-tools.preinst

```diff
@@ -11,12 +11,63 @@ remove_old_config_fields() {
     fi
 }
 
+restore_previous_conffile() {
+    # Back up existing conffile in case of an error unwind
+    cp -a /etc/ubuntu-advantage/uaclient.conf /etc/ubuntu-advantage/uaclient.conf.preinst-remove
+    # Restore the default conffile that shipped with 27.11.3 through 27.12
+    cat > /etc/ubuntu-advantage/uaclient.conf <<EOT
+# Ubuntu Pro Client config file.
+# If you modify this file, run "pro refresh config" to ensure changes are
+# picked up by Ubuntu Pro Client.
+
+contract_url: https://contracts.canonical.com
+data_dir: /var/lib/ubuntu-advantage
+log_file: /var/log/ubuntu-advantage.log
+log_level: debug
+security_url: https://ubuntu.com/security
+timer_log_file: /var/log/ubuntu-advantage-timer.log
+daemon_log_file: /var/log/ubuntu-advantage-daemon.log
+ua_config:
+  apt_http_proxy: null
+  apt_https_proxy: null
+  http_proxy: null
+  https_proxy: null
+  update_messaging_timer: 21600
+  update_status_timer: 43200
+  metering_timer: 14400
+EOT
+}
+
 case "$1" in
     install|upgrade)
         if [ -n "$2" ]; then
             PREVIOUS_PKG_VER=$2
             remove_old_config_fields "$PREVIOUS_PKG_VER"
         fi
+    
+        # LP: #2003977
+        # If the user used "pro config set apt_news=false|true" previously,
+        # then we don't want a conffile prompt if they haven't otherwise
+        # changed the conffile. In these two cases, restore the conffile back
+        # to how it shipped, and then fix up in postinst if required. The
+        # version gate is open ended, and should be closed when the apt_news
+        # configuration is moved away from a conffile.
```

Comment:
Has this been tested for the case when the user has already upgraded to the version of the package currently in -updates, and rejected the conffile update?  In that case, the "expected" conffile hash will not match that from 27.11.3, and resetting the file to the previous state will not clear the prompt the *next* time the conffile changes; you would want to reset the file to the pristine contents of the 27.13.1 version.

Reply:
Yes, we understand this issue and we will address this in the next release to check the md5sum for the one we deliver for 27.13.1 to prevent exactly that. But we understand it should not be a blocker for this release. But let us know if that assumption is wrong

Reply:
I'm aware that `restore_previous_conffile` will need adjusting in the subsequent upload after this one to restore the more recent conffile. Is this what you mean? I didn't implement this initially as it's tricky to test, but have tried to make it clear in surrounding docs (in my https://bugs.launchpad.net/ubuntu/+source/ubuntu-advantage-tools/+bug/2003977/comments/5) that will be a hard requirement to adjust `preinst` to restore the correct conffile based on `$2` in the next upload after this.